apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -64,4 +66,37 @@ export const folderRouter = router({
         parentId,
       }
     }),
+  editFolder: protectedProcedure
+    .input(editFolderSchema)
+    .mutation(async ({ input: { resourceId, permalink, title, siteId } }) => {
+      if (!title && !permalink) {
+        return
+      }
+
+      return db.transaction().execute(async (tx) => {
+        let baseQuery = tx
+          .updateTable("Resource")
+          .where("Resource.id", "=", resourceId)
+          .where("Resource.siteId", "=", Number(siteId))
+          .where("Resource.type", "=", "Folder")
+
+        if (permalink) {
+          baseQuery = baseQuery.set({
+            permalink,
+          })
+        }
+
+        if (title) {
+          baseQuery = baseQuery.set({
+            title,
+          })
+        }
+
+        await baseQuery.execute()
+        return db
+          .selectFrom("Resource")
+          .where("Resource.id", "=", resourceId)
+          .executeTakeFirstOrThrow()
+      })
+    }),
```

Comment:
You don't need the conditional, any undefined assignment is stripped. You can also return directly after updating the row with `RETURNING`. This removes the need for a transaction (and multiple queries). Also suggest returning the default select instead of... nothing. 
Actually that will throw runtime error since its invalid SQL. But kysely cannot catch.

```suggestion
        return await db
          .updateTable("Resource")
          .where("Resource.id", "=", resourceId)
          .where("Resource.siteId", "=", Number(siteId))
          .where("Resource.type", "=", "Folder")
          .set({
            permalink,
            title
          })
          .returning(defaultFolderSelect)
          .execute()
```